src/utils/FeatureDetectionUtils.ts

```diff
@@ -1,3 +1,13 @@
+// Necessary because of https://github.com/Microsoft/TypeScript/issues/14062
+// Introduced in typescript 2.2, should be fixed in 2.3
+interface IDocument {
+  implementation: {
+    hasFeature: any;
+  };
+}
+
+declare var document: IDocument;
+
 export class FeatureDetectionUtils {
```

Comment:
If it's not used anywhere else.... Remove it.

I would remove the file completely, and then we don't need this fix.